tools/api/generate_go_protobuf.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python
+from subprocess import check_output
+from subprocess import call
+
+import glob
+import os
+import shutil
+import sys
+
+# Find the locations of the workspace root and the generated files directory.
+workspace = check_output(['bazel', 'info', 'workspace']).strip()
+bazel_bin = check_output(['bazel', 'info', 'bazel-bin']).strip()
+targets = '@envoy_api//...'
+import_base = 'github.com/envoyproxy/data-plane-api/api'
+output_base = 'go_out'
+
+go_protos = check_output([
+    'bazel',
+    'query',
+    'kind("go_proto_library", %s)' % targets,
+]).split()
+
+# Each rule has the form @envoy_api//foo/bar:baz_go_proto.
+# First build all the rules to ensure we have the output files.
+if call(['bazel', 'build'] + go_protos) != 0:
+  print('Build failed')
+  sys.exit(1)
+
+shutil.rmtree(os.path.join(workspace, output_base, 'envoy'), ignore_errors=True)
+for rule in go_protos:
+  # Example rule:
+  # @envoy_api//envoy/config/bootstrap/v2:pkg_go_proto
+  #
+  # Example generated directory:
+  # bazel-bin/external/envoy_api/envoy/config/bootstrap/v2/linux_amd64_stripped/pkg_go_proto%/github.com/envoyproxy/data-plane-api/api/envoy/config/bootstrap/v2/
+  #
+  # Example output directory:
+  # go_out/envoy/config/bootstrap/v2
+  rule_dir, proto = rule[len("@envoy_api//"):].rsplit(':', 1)
+
+  input_dir = os.path.join(bazel_bin, 'external', 'envoy_api', rule_dir, 'linux_amd64_stripped',
```

Comment:
This code is quite similar to https://github.com/envoyproxy/envoy/blob/master/docs/build.sh#L66. Is there a way to refactor to share, or maybe you can avoid reaching into `amd64` named dirs somehow like we do for docs?

Reply:
linux_amd64_stripped is specific to rules_go I think. It's not coming from bazel itself. I've tried to figure out how to get generated files with bazel query but eventually gave up.

Reply:
Yes, it's due to compilation modes in go. It statically links everything, so it has to keep outputs in sub-directories https://github.com/bazelbuild/rules_go/blob/0303b3a69695e35940b09ddbf7a444bcc7fbefd4/go/private/mode.bzl.

Reply:
Can you specify `--strip=always` or `-c fastbuild` when `bazel build` is invoked in line 25? Otherwise the directory name is interfered by .bazelrc (e.g. I have `-c dbg` in my bazelrc)

Reply:
Maybe it's better to recommend running this script with empty `--bazelrc=/dev/null` on linux amd64?
If you poke around the logic for the choice of the directory, it's pretty convoluted and interferes with msan/race/etc: https://github.com/bazelbuild/rules_go/blob/0303b3a69695e35940b09ddbf7a444bcc7fbefd4/go/private/mode.bzl#L135

Reply:
yeah an empty bazelrc works too but I think that ignores workspace .bazelrc

Reply:
ok, -c fastbuild is better than ignoring workspace bazel rc.